File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-1.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        i = 1
        j = 1
        while i<len(nums):
            while i<len(nums) and nums[i]==nums[i-1]:
                i+=1
            if i==len(nums) :
                break
            nums[j]=nums[i]
            j+=1
            i+=1
        return j

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.rs
impl Solution {
    pub fn find_min(nums: Vec<i32>) -> i32 {
        let mut lo = 0;
        let mut hi = nums.len()-1;
        let mut mid = 0;
        while lo<hi {
            if nums[lo]<=nums[hi] {
                return nums[lo] as i32;
            }
            mid = (lo+hi) >> 1;
            if nums[mid] > nums[hi] {
                lo = mid + 1
            }
            else {
                hi = mid
            }
        }
        nums[lo]
    }
}
